src/adapter/graphql/src/mutations/dataset_metadata_mut.rs

```diff
@@ -58,11 +60,25 @@ impl DatasetMetadataMut {
     }
 
     /// Updates or clears the dataset readme
+    #[graphql(guard = "LoggedInGuard::new()")]
+    // TODO: guard should include permission check
     async fn update_readme(
         &self,
         ctx: &Context<'_>,
         content: Option<String>,
     ) -> Result<UpdateReadmeResult> {
+        match check_dataset_write_access(ctx, &self.dataset_handle).await {
```

Comment:
This is a lot of code ... would suck for this to spread in to every mutation.

Let's think (outside of this PR) how we can push these permission checks down to the dataset repo layer (e.g. one idea would be to separate `Dataset` interface into "read" and "write" parts).

Reply:
It's the same problem as we had with authorization checks at CLI commands level.